contracts/SL2RD.sol

```diff
@@ -475,4 +481,29 @@ contract SL2RD is
         require(asset.owner() == address(this), "SHARE025");
         asset.transferOwnership(msg.sender);
     }
+
+    /// @notice Sets the ERC20 contract address (e.g., for USDC payments).
+    function setERC20ContractAddress(
+        address contractAddress_
+    ) public override afterInit onlyOwner nonReentrant {
+        require(
+            contractAddress_ != address(0),
+            "SHARE001"
```

Comment:
@xiangformless This is the right idea but the error code is wrong. You'll need to make a new one specific to this error.